activities/views/posts.py

```diff
@@ -203,28 +207,53 @@ def clean_text(self):
 
     def get_initial(self):
         initial = super().get_initial()
-        initial[
-            "visibility"
-        ] = self.request.identity.config_identity.default_post_visibility
-        if self.reply_to:
-            initial["reply_to"] = self.reply_to.pk
-            initial["visibility"] = self.reply_to.visibility
-            initial["text"] = f"@{self.reply_to.author.handle} "
+        if self.post_obj:
+            initial.update(
+                {
+                    "reply_to": self.reply_to.pk if self.reply_to else "",
+                    "visibility": self.post_obj.visibility,
+                    "text": self.post_obj.content,
+                    "content_warning": self.post_obj.summary,
+                }
+            )
+        else:
+            initial[
+                "visibility"
+            ] = self.request.identity.config_identity.default_post_visibility
+            if self.reply_to:
+                initial["reply_to"] = self.reply_to.pk
+                initial["visibility"] = self.reply_to.visibility
+                initial["text"] = f"@{self.reply_to.author.handle} "
         return initial
 
     def form_valid(self, form):
-        post = Post.create_local(
-            author=self.request.identity,
-            content=form.cleaned_data["text"],
-            summary=form.cleaned_data.get("content_warning"),
-            visibility=form.cleaned_data["visibility"],
-            reply_to=self.reply_to,
-        )
-        # Add their own timeline event for immediate visibility
-        TimelineEvent.add_post(self.request.identity, post)
+        post_id = form.cleaned_data.get("id")
+        if post_id:
+            post = get_object_or_404(self.identity.posts, pk=post_id)
+            post.content = form.cleaned_data["text"]
+            post.summary = form.cleaned_data.get("content_warning")
+            post.visibility = form.cleaned_data["visibility"]
+            post.save()
+
+            # Should there be a timeline event for edits?
+            # E.g. "@user edited #123"
```

Comment:
Should there be a timeline event for edits? If not, do we want to indicate on the post view that it has been edited?

Reply:
Timeline event no, marked on post yes. The `edited` field on the Post object is intended to be this flag.